ethsigner/signer/multikey/src/test/java/tech/pegasys/ethsigner/signer/multikey/MultiKeyTransactionSignerProviderTest.java

```diff
@@ -86,8 +89,20 @@ void getSignerForAvailableMetadataReturnsSigner() {
 
   @Test
   void getAddresses() {
-    Collection<SigningMetadataFile> files = Collections.singleton(metadataFile);
+    final Collection<SigningMetadataFile> files = Collections.singleton(metadataFile);
     when(loader.loadAvailableSigningMetadataTomlConfigs()).thenReturn(files);
     assertThat(signerFactory.availableAddresses()).containsExactly("0x" + LOWERCASE_ADDRESS);
   }
+
+  @Test
+  void signerIsLoadedSuccessfullyWhenAddressHasCaseMismatchToFilename() throws URISyntaxException {
+    final FileBasedSigningMetadataFile capitalisedMetadata =
+        new FileBasedSigningMetadataFile(LOWERCASE_ADDRESS.toUpperCase() + ".toml",
+            Path.of(Resources.getResource("metadata-toml-configs").toURI()).resolve(KEY_FILENAME),
+            Path.of(Resources.getResource("metadata-toml-configs").toURI()).resolve(PASSWORD_FILENAME));
+
+    final TransactionSigner signer = signerFactory.createSigner(capitalisedMetadata);
+    assertThat(signer).isNotNull();
+    assertThat(capitalisedMetadata.getBaseFilename()).isNotEqualTo(signer.getAddress().substring(2));
```

Comment:
thin we need an assertion that signer that was created has the lower case address

Reply:
I don't THINK so ... do we care about what the string rep of a key is?

Reply:
ah - ok - added a few asserts to confirm the signer loaded is that which was requested.